packages/gatsby/src/components/link.js

```diff
@@ -0,0 +1,189 @@
+/*global __PREFIX_PATHS__, __PATH_PREFIX__ */
+import React from "react"
```

Comment:
Rather then moving the source code here it might make sense to leave the external pacakge and have `gatsby` depend on it directly and re-export

Reply:
Hmm...quite not sure about this. If there are plans to have more components in `gatsby` package, I think it's better to have the code here as we may not want other components to be in their own packages. If that's not something we're looking at now, yeah, it makes sense to just leave the package as is and re-export the component.

Reply:
it might be the case taht new components live here, but I do think there is probably less friction to re-export. The other benefit tho is that someone can update/bug fix gatsby-link without requiring a gatsby release. That said I defer to what @KyleAMathews thinks

Reply:
I agree with @jquense. It's a much smaller change to leave it where it is and just export it. Also it keeps our packaging setup simpler as with gatsby-link we know it's browser code and we can build and publish it accordingly.

Reply:
> The other benefit tho is that someone can update/bug fix gatsby-link without requiring a gatsby release.
@jquense,  might be a silly question, but I don't quite understand this - do you mean we'll keep publishing new changes to `gatsby-link` on to npm as a separate package without publishing `gatsby`? If we want `Link` to be imported from `gatsby`, we've to discontinue publishing a separate package no?

Reply:
@tsriram no we'd still publish gatsby-link as a separate package — it'd just now be a dependency of `gatsby` and we'd instruct people to use the re-exported version there.

@jquense lerna would actually still publish `gatsby` on every new `gatsby-link` change — excess of caution perhaps but whatever. We already release gatsby a few times a day or so so no big deal :-)

Reply:
@KyleAMathews Cool. `gatsby-link` already seems to be a dependency of `gatsby`: https://github.com/gatsbyjs/gatsby/blob/fb6313ef328ac77773d49b16963fea310d892ab6/packages/gatsby/package.json#L52

So only re-exporting needs to be done here I guess.

Reply:
Also updating example sites and www and any docs referencing gatsby-link :-)

Reply:
Oh yeah :smile: